src/engine/runtime.mjs

```diff
@@ -577,6 +605,223 @@ export default class Runtime extends EventEmitter {
         });
     }
 
+
```

Comment:
I am hesitant to commit threading-related code to the patch VM as our use of threads is different enough from Scratch's that we will have to use our own implementation. Scatch's thread implementation is closely coupled with their sequencer class. We will not be using the sequencer so it doesn't make sense to commit any thread-related code right now either. That being said, we probably do need to have a conversation about how events will work.